gax-httpjson/src/main/java/com/google/api/gax/httpjson/ProtoMessageRequestFormatter.java

```diff
@@ -91,19 +102,56 @@ public String getPath(RequestT apiMessage) {
     return pathTemplate.instantiate(pathVarsExtractor.extract(apiMessage));
   }
 
+  @BetaApi
+  @Override
+  public List<PathTemplate> getAdditionalPathTemplates() {
+    return additionalPathTemplates;
+  }
+
   /* {@inheritDoc} */
   @Override
   public PathTemplate getPathTemplate() {
     return pathTemplate;
   }
 
+  @Override
```

Comment:
Do we have to override hashcode() and equals() methods for this class? I think if you don't compare them or put them in a collection, then you don't have to.

Reply:
There is `toBuilder()` semantics in this class, so I added `equals()`/`hashCode()` to do testing of that logic, but seems like adding equals/hashcode added it own problems of branch coverage. I guess I'll just remove those as they cause more trouble than they should.